jax/_src/lib/xla_bridge.py

```diff
@@ -297,21 +287,32 @@ def backends():
   with _backend_lock:
     if _backends:
       return _backends
-
-    if FLAGS.jax_platforms:
-      jax_platforms = FLAGS.jax_platforms.split(",")
+    default_priority = -1000
+    if config.jax_platforms:
+      jax_platforms = config.jax_platforms.split(",")
       platforms = []
       # Allow platform aliases in the list of platforms.
       for platform in jax_platforms:
         platforms.extend(expand_platform_alias(platform))
       priorities = range(len(platforms), 0, -1)
       platforms_and_priorites = zip(platforms, priorities)
-    else:
-      platforms_and_priorites = (
-          (platform, priority) for platform, (_, priority)
-          in _backend_factories.items())
+      for platform, priority in platforms_and_priorites:
```

Comment:
Instead of creating new logic for when jax_platforms instead, please combine with the existing logic starting around L316. The only difference should be how to handle backend initialization errors, so you can create a new variable like `continue_on_backend_init_errors` to keep track of what behavior should be used. This logic is already a bit different from the below logic (e.g. doesn't handle the "rocm" GPU stuff).